packages/financial-templates-lib/src/price-feed/BasketSpreadPriceFeed.js

```diff
@@ -112,6 +113,23 @@ class BasketSpreadPriceFeed extends PriceFeedInterface {
     return this._getSpreadFromBasketPrices(experimentalPrices, baselinePrices, denominatorPrice);
   }
 
+  debugHistoricalData(time) {
+    let priceFeedErrorDetails = [];
+    this.allPriceFeeds.map(priceFeed => {
+      const hasPrice = priceFeed.getHistoricalPrice(time);
+      if (!hasPrice) {
+        if (priceFeed instanceof MedianizerPriceFeed) {
+          // MedianizerPriceFeed.debugHitoricalData returns an array of error details so we will
+          // concat its error log array.
+          priceFeedErrorDetails = priceFeedErrorDetails.concat(priceFeed.debugHistoricalData(time));
```

Comment:
Why not just have every price feed return a string as its own error message and higher-level price feeds just join the strings with separators so you don't have to detect the medianizer price feed specifically?

Reply:
This is a good idea, originally I wanted to shape the errors as an array of strings, but I don't think the array of strings vs concatenated string makes much difference to the end user.

Reply:
However, I still like the idea of the higher level price feeds defining the error strings based on the lower level price feed's UUID's. Ultimately, this approach removes the need for each lower level price feed to implement `debugHistoricalData`, but the tradeoff is that the higher level pricefeeds should check if a lower level price feed has implemented `debugHistoricalData` first before it constructs an error string for the lower price feed. Look at [this commit](https://github.com/UMAprotocol/protocol/pull/2426/commits/be57391633074bd8769656769ded7740ade7f685) for example.